rosbag2_transport/include/rosbag2_transport/play_options.hpp

```diff
@@ -101,6 +104,8 @@ struct PlayOptions
   bool disable_loan_message = false;
 };
 
+void declare_play_options_rw_params(std::shared_ptr<rclcpp::Node> nh, PlayOptions & po);
```

Comment:
I'm of a mixed mind with these free functions. I think I'd like to keep all of them out of the API-space of the configuration structs themselves. Have one central place where all Parameters are handled and transformed into the C++ API configuration structs, and keep the struct definition location agnostic of how those structs might be created.

The alternative argument is to keep related things together so they are less likely to get out of sync somehow, but I'm having a hard time convincing myself that it's better than my above suggestion.